backend/app/controllers/api/favorites_controller.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Api
+  # Favorites Controller class
+  class FavoritesController < ApplicationController
+    skip_before_action :verify_authenticity_token
+    before_action :require_logged_in
+
+    def create
+      @favorite = Favorite.new(user_id: current_user.id, restaurant_id: params[:restaurant_id])
+
+      if @favorite.save
+        render :show
+      else
+        render json: @favorite.errors.full_messages, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @favorite = Favorite.find_by(user_id: current_user.id, restaurant_id: params[:restaurant_id])
+
+      if @favorite
+        @favorite.destroy
```

Comment:
Should this be `destroy!`? Seems worth failing hard if it can't happen for some reason.